pkg/nsx/services/vpc/vpc.go

```diff
@@ -718,9 +718,15 @@ func (s *VPCService) IsLBProviderChanged(existingVPC *model.Vpc, lbProvider LBPr
 		return false
 	}
 	if lbProvider == AVILB {
+		if existingVPC.LoadBalancerVpcEndpoint == nil {
+			log.Info("AVI LoadBalancerVpcEndpoint is nil, will result in change", "VPC path", *existingVPC.Path)
+			return true
+		}
 		if existingVPC.LoadBalancerVpcEndpoint.Enabled == nil || !*existingVPC.LoadBalancerVpcEndpoint.Enabled {
+			log.Info("AVI LoadBalancerVpcEndpoint is disabled or nil, will result in change", "VPC path", *existingVPC.Path)
 			return true
 		}
```

Comment:
if existingVPC.LoadBalancerVpcEndpoint == nil || existingVPC.LoadBalancerVpcEndpoint.Enabled == nil || !*existingVPC.LoadBalancerVpcEndpoint.Enabled {
    log.Info("Need to update AVI LoadBalancerVpcEndpoint")
    return true
}

Reply:
ok